pennylane_lightning/core/src/simulators/lightning_kokkos/algorithms/tests/Test_AdjointJacobianKokkos.cpp

```diff
@@ -103,6 +103,82 @@ TEMPLATE_PRODUCT_TEST_CASE(
     REQUIRE(isApproxEqual(sv1.getDataVector(), sv2.getDataVector(), eps));
 }
 
+TEMPLATE_PRODUCT_TEST_CASE(
+    "AdjointJacobianKokkos::adjointJacobian Op=Controlled-Mixed, Obs=[XXX]",
+    "[AdjointJacobianKokkos]", (StateVectorKokkos), (float, double)) {
+    using StateVectorT = TestType;
+    using PrecisionT = typename StateVectorT::PrecisionT;
+    AdjointJacobian<StateVectorT> adj;
+    std::vector<PrecisionT> param{-M_PI / 7, M_PI / 5, 2 * M_PI / 3};
+    std::vector<std::size_t> tp{0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+    {
+        const std::size_t num_qubits = 3;
+        const std::size_t num_obs = 1;
+        std::vector<PrecisionT> jacobian(num_obs * tp.size(), 0);
+
+        StateVectorT psi(num_qubits);
+
+        const auto obs = std::make_shared<TensorProdObs<StateVectorT>>(
+            std::make_shared<NamedObs<StateVectorT>>(
+                "PauliX", std::vector<std::size_t>{0}),
+            std::make_shared<NamedObs<StateVectorT>>(
+                "PauliX", std::vector<std::size_t>{1}),
+            std::make_shared<NamedObs<StateVectorT>>(
+                "PauliX", std::vector<std::size_t>{2}));
+        auto ops = OpsData<StateVectorT>(
+            {"RZ", "RY", "RZ", "CNOT", "CNOT", "RX", "RY", "RZ",
+             "SingleExcitation", "RZ", "RY", "RZ"},
+            {{param[0]},
+             {param[1]},
+             {param[2]},
+             {},
+             {},
+             {param[0]},
+             {param[1]},
+             {param[2]},
+             {param[0]},
+             {param[0]},
+             {param[1]},
+             {param[2]}},
+            {{0},
+             {0},
+             {0},
+             {0, 1},
+             {1, 2},
+             {1},
+             {2},
+             {1},
+             {0, 2},
+             {1},
+             {1},
+             {1}},
+            {false, false, false, false, false, false, false, false, false,
+             false, false, false},
+            {{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}},
+            {{}, {}, {}, {}, {}, {0}, {0}, {0}, {1}, {}, {}, {}},
+            {{}, {}, {}, {}, {}, {true}, {true}, {true}, {true}, {}, {}, {}});
+
+        JacobianData<StateVectorT> tape{
+            param.size(), psi.getLength(), psi.getData(), {obs}, ops, tp};
+
+        adj.adjointJacobian(std::span{jacobian}, tape, psi, true);
+
+        CAPTURE(jacobian);
+
+        // Computed with PennyLane using default.qubit.adjoint_jacobian
+        CHECK(0.0 == Approx(jacobian[0]).margin(1e-7));
```

Comment:
can we set the margin with a macro?

Reply:
I"d like to get rid of these margins all at once as they seem too repetitive! If we could set the default to `1e-6`, we can remove all these tolerance values.